docs/posts/ibis-analytics/index.qmd

```diff
@@ -1,341 +1,1969 @@
 ---
 title: Modern, hybrid, open analytics
-author: "Cody Peterson"
-date: "2023-09-02"
+author: "Cody"
+date: "2024-01-20"
 image: "thumbnail.png"
 categories:
     - blog
     - duckdb
+    - bigquery
     - case study
-    - dogfood
-draft: true
 ---
 
 ## Introduction
 
-As a Python data user, I've wanted a more modular, composable, and scalable ecosystem. *It finally feels close*. Wes McKinney released pandas c. 2009 to bring dataframes into Python and it became one of the most used packages. It was built when small data was smaller and though still an excellent dataframe library today has some downsides [Wes wrote about in his "Apache Arrow and the '10 things I hate about pandas'" blog post](https://wesmckinney.com/blog/apache-arrow-pandas-internals). Wes created Ibis (and Apache Arrow) to address these issues and more with a different approach than any other Python dataframe library by supporting any backend, including pandas or DuckDB or Polars or BigQuery or Snowflake or PySpark or over a dozen others.
-
-I've experienced many pains of the ecosystem myself. I downloaded a 1 GB CSV from a database GUI and used pandas to munge it because I couldn't figure it out in SQL. When building larger ML demos that included data preprocessing on GBs, I hit OOM errors and learned about how much data expands in memory and why to store files in compressed parquet and how to partition data. I started using PySpark and Dask to overcome hurdles. I slowly learned about databases. Fortunately while I've been learning, others ([including many Ibis contributors](https://github.com/ibis-project/ibis/contributors)) have been building a modern, open-source Python data ecosystem!
-
-Ibis plays a part but is only one component in any real data project. We'll look at how to combine a variety of open-source tools and freemium services including:
+As a Python data user, I've wanted a more modular, composable, and scalable
+ecosystem. *I think it's here*. Wes McKinney released pandas c. 2009 to bring
+dataframes into Python and it became one of the most used software packages. It
```

Comment:
```suggestion
the power of SQL to Python via the DataFrame and it became one of the most used software packages. It
```

I think this may appeal to a slightly more general audience.

Reply:
this is on pandas? it didn't use/bring the power of SQL?

Reply:
Literally no, but it brought the concepts (and the conceptual power) of SQL to the forefront of the Python ecosystem.